Extension package management enhancement.

BrainforgeUK opened the pull request:
See issue #17579 - Managing extension packages
https://issues.joomla.org/tracker/joomla-cms/17579

Pull Request for Issue # .

### Summary of Changes
Added -- Select Package -- list to search tool in extension manager
Added links to plugin / template edit in extension manager list display



### Testing Instructions
Goto .../administrator/index.php?option=com_installer&view=manage
Go to Search Tools and Select a Package



### Expected result



### Actual result



### Documentation Changes Required

rigin committed:
Add a note field from the #__fields_groups table to the field list query

brianteeman commented on administrator/components/com_installer/models/manage.php:
```diff
@@ -302,6 +302,7 @@ protected function getListQuery()
 		$type     = $this->getState('filter.type');
 		$clientId = $this->getState('filter.client_id');
 		$folder   = $this->getState('filter.folder');
+		$packageId  = $this->getState('filter.package_id');
```
Please can you adjust the spaces on the lines above so that all the = are in allignment

BrainforgeUK replied:
Thanks, should have done that earlier.
See zero-24 done it already - got there before me!

Quy replied:
Remove the extra space before `=`

zero-24 commented:
@brainforgeUK Thanks for your PR. i have just fixed some minor code style issues. I hope i get some time to test this feature too. If you have any questions about the changes I made please feel free to ask me.

Quy commented on administrator/language/en-GB/en-GB.com_installer.ini:
```diff
@@ -264,6 +264,7 @@ COM_INSTALLER_VALUE_FOLDER_NONAPPLICABLE="N/A"
 COM_INSTALLER_VALUE_FOLDER_SELECT="- Select Folder -"
 COM_INSTALLER_VALUE_STATE_SELECT="- Select Status -"
 COM_INSTALLER_VALUE_TYPE_SELECT="- Select Type -"
+COM_INSTALLER_VALUE_PACKAGE_SELECT="- Select Package -"
```
Alpha order. Move it after line 264.

Quy commented on administrator/components/com_installer/views/manage/tmpl/default.php:
```diff
@@ -100,7 +100,21 @@
+											<a href="index.php?option=com_plugins&task=plugin.edit&extension_id=<?php echo $item->extension_id; ?>">
+												<?php echo $item->name; ?>
+											</a>
+										<?php break; ?>
```
Indent

Quy commented on administrator/components/com_installer/helpers/installer.php:
```diff
@@ -184,4 +184,32 @@ public static function getStateOptions()
 	}
+
+	/**
+	 * Get a list of filter options for the pakcages.
```
Spelling `pakcages`

BrainforgeUK replied:
Changed.

Quy commented on administrator/components/com_installer/views/manage/tmpl/default.php:
```diff
@@ -100,7 +100,21 @@
+										<a href="index.php?option=com_plugins&task=plugin.edit&extension_id=<?php echo $item->extension_id; ?>">
+										    <?php echo $item->name; ?>
+										</a>
+									<?php break; ?>
```
Indent this line and dedent the following line.

BrainforgeUK replied:
My editor setting replaces tabs with spaces.
Seems to be causing confusion!

zero-24 commented on administrator/components/com_installer/views/manage/tmpl/default.php:
```diff
@@ -100,21 +100,21 @@
 						<td>
 							<label for="cb<?php echo $i; ?>">
 								<span class="bold hasTooltip" title="<?php echo JHtml::_('tooltipText', $item->name, $item->description, 0); ?>">
-									<?php switch($item->type) :
```
why did you replace the tabs by spaces?

zero-24 commented:
@brainforgeUK it looks like you have mix up tabs and spaces in the last commit again ;)

akritianand commented:
Hey! After applying your patch all I see is a blank input field in the search tools. Entering text data does nothing in it.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/17585">issues.joomla.org/tracker/joomla-cms/17585</a>.</sub>

zero-24 commented:
@brainforgeUK i can confirm the result of @akritianand 

To me it looks like this in the backend.
![image](https://user-images.githubusercontent.com/2596554/29484732-132347ea-84c5-11e7-9392-ff08e0422f43.png)

BrainforgeUK commented:
Fixed - package field class file missing!
New file got left out of git by accident.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/17585">issues.joomla.org/tracker/joomla-cms/17585</a>.</sub>

BrainforgeUK commented:
Be better to change the package field class in package.php into a standard form field.
i.e. move
**from :**
`/administrator/components/com_installer/models/fields/package.php`
**to :**
`/libraries/joomla/form/fields/package.php`

Can then add package as a plugins list search  - it makes sense to distribute several plugins as a package.
`administrator/index.php?option=com_plugins`
Other people may find the same form field useful elsewhere.

Will implement above shortly!

zero-24 commented on administrator/components/com_installer/views/manage/tmpl/default.php:
```diff
@@ -17,128 +17,152 @@
+              <th>
+                <?php echo JHtml::_('searchtools.sort', 'COM_INSTALLER_HEADING_LOCATION', 'client_translated', $listDirn, $listOrder); ?>
+              </th>
+              <th>
```
again tabs and spaces ;)

zero-24 commented on libraries/joomla/form/fields/package.php:
```diff
@@ -0,0 +1,108 @@
+   * The form field package.
+   *
+   * @var    string
+   * @since  3.8
```
same in this files. Spaces & Tabs.

please also replace the fixed version numbers with `__DEPLOY_VERSION__` ;)

Heggi93 commented:
I have tested this item :white_check_mark: successfully on acb71211a133fed3c1638d6d2c127d4a03dfb71b<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/17585">issues.joomla.org/tracker/joomla-cms/17585</a>.</sub>

sanderpotjer commented:
I have tested this item :red_circle: unsuccessfully on acb71211a133fed3c1638d6d2c127d4a03dfb71b<br /><br />@brainforgeUK nice addition! The functionality is working correctly, but some code style improvements are still needed (tabs & spaces) as commented by others before.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/17585">issues.joomla.org/tracker/joomla-cms/17585</a>.</sub>

BrainforgeUK commented:
We seem to have got stuck with this.
Not in Joomla 4.0
How do we move on?

Quy commented:
@brainforgeUK Please fix coding style as mentioned previously. Use tabs (not spaces) to indent code.

Quy commented on administrator/components/com_plugins/models/forms/filter_plugins.xml:
```diff
@@ -36,6 +36,15 @@
 		</field>
+
+		<field
+				name="package_id"
```
lines 41-44 Fix/remove tabs.

BrainforgeUK replied:
Changed. Thanks for pointing out.

Quy commented on libraries/joomla/form/fields/packagelist.php:
```diff
@@ -0,0 +1,108 @@
+	/**
+	 * Only packages containing extensions of specified type
+	 *
+	 * @var    float
```
wrong type

BrainforgeUK replied:
Changed. Thanks for pointing out.

Quy commented on libraries/joomla/form/fields/packagelist.php:
```diff
@@ -0,0 +1,108 @@
+	/**
+	 * Only packages with the given value of enabled field
+	 *
+	 * @var    float
```
wrong type

BrainforgeUK replied:
Changed. Thanks for pointing out.

Quy commented on libraries/joomla/form/fields/packagelist.php:
```diff
@@ -0,0 +1,108 @@
+
+		if ($return)
+		{
+			$this->extensiontype   = isset($this->element['extensiontype']) ? $this->element['extensiontype'] : null;
```
lines 69-70 remove extra space before =

BrainforgeUK replied:
Changed. Thanks for pointing out.

Quy commented on libraries/joomla/form/fields/packagelist.php:
```diff
@@ -0,0 +1,108 @@
+			    ->from('#__extensions AS a')
+			    ->where('a.type = ' . $db->quote('package'))
+			    ->order('a.name');
+		if (!empty($this->extensiontype)) {
```
Move `{` to the next line and add a blank line above `if`.

BrainforgeUK replied:
Changed. Thanks for pointing out.

Quy commented on libraries/joomla/form/fields/packagelist.php:
```diff
@@ -0,0 +1,108 @@
+		if (!empty($this->extensiontype)) {
+			$query->where('EXISTS ( SELECT NULL FROM #__extensions AS b WHERE b.package_id = a.extension_id AND b.type = ' . $db->quote($this->extensiontype) . ')');
+		}
+		if (strlen($this->packageenabled)) {
```
Move `{` to the next line and add a blank line above `if`.

BrainforgeUK replied:
Changed. Thanks for pointing out.

Quy commented on libraries/joomla/form/fields/packagelist.php:
```diff
@@ -0,0 +1,108 @@
+		if (strlen($this->packageenabled)) {
+			$query->where('a.enabled = ' . $db->quote($this->packageenabled));
+		}
+		$db->setQuery($query);
```
Add a blank line above.

BrainforgeUK replied:
Changed. Thanks for pointing out.

Quy commented on administrator/components/com_installer/views/manage/tmpl/default.php:
```diff
@@ -100,7 +100,31 @@
+							    // No link to current search package
+							    break;
+							}
+							$href='index.php?option=com_installer&view=manage&filter[package_id]=' . $item->extension_id;
```
Add space before and after `=`

BrainforgeUK replied:
Changed. Thanks for pointing out.

Quy commented on administrator/components/com_installer/views/manage/tmpl/default.php:
```diff
@@ -100,7 +100,31 @@
+							$href='index.php?option=com_installer&view=manage&filter[package_id]=' . $item->extension_id;
+							break;
+						case 'plugin':
+							$href='index.php?option=com_plugins&task=plugin.edit&extension_id=' . $item->extension_id;
```
Add space before and after `=`

BrainforgeUK replied:
Changed. Thanks for pointing out.

Quy commented on administrator/components/com_installer/views/manage/tmpl/default.php:
```diff
@@ -100,7 +100,31 @@
+							$href='index.php?option=com_plugins&task=plugin.edit&extension_id=' . $item->extension_id;
+							break;
+						case 'template':
+							$href='index.php?option=com_templates&view=template&id=' . $item->extension_id;
```
Add space before and after `=`

BrainforgeUK replied:
Changed. Thanks for pointing out.

Quy commented on administrator/components/com_installer/views/manage/tmpl/default.php:
```diff
@@ -100,7 +100,31 @@
+							$href='index.php?option=com_templates&view=template&id=' . $item->extension_id;
+							break;
+					}
+					if ($href !== null) {
```
Move `{` to next line.

BrainforgeUK replied:
Changed. Thanks for pointing out.

Quy commented on administrator/components/com_installer/views/manage/tmpl/default.php:
```diff
@@ -100,7 +100,31 @@
+						echo '<a href="' . $href . '">';
+					}
+					echo $item->name;
+					if ($href !== null) {
```
Move `{` to next line.

BrainforgeUK replied:
Changed. Thanks for pointing out.

Quy commented on administrator/components/com_installer/views/manage/tmpl/default.php:
```diff
@@ -100,7 +100,31 @@
-									<?php echo $item->name; ?>
+					<?php
+					$href = null;
+					switch($item->type) {
```
Add space after `switch` and move `{` to next line.

BrainforgeUK replied:
Changed. Thanks for pointing out.

Quy commented on administrator/components/com_installer/views/manage/tmpl/default.php:
```diff
@@ -100,7 +100,31 @@
+					$href = null;
+					switch($item->type) {
+						case 'package':
+							if (!empty($item->search_package_id) && $item->search_package_id == $item->extension_id) {
```
Move `{` to next line.

BrainforgeUK replied:
Changed. Thanks for pointing out.

Quy commented:
For your reference on coding style: https://developer.joomla.org/coding-standards/php-code.html

Quy commented on administrator/components/com_installer/views/manage/tmpl/default.php:
```diff
@@ -100,7 +100,34 @@
+					switch ($item->type)
+					{
+						case 'package':
+							if (!empty($item->search_package_id) && $item->search_package_id == $item->extension_id) {
```
Move `{` to next line. Use tab instead of spaces in the following 2 lines.

Quy commented on administrator/components/com_installer/views/manage/tmpl/default.php:
```diff
@@ -100,7 +100,34 @@
 								<span class="bold hasTooltip" title="<?php echo JHtml::_('tooltipText', $item->name, $item->description, 0); ?>">
-									<?php echo $item->name; ?>
+					<?php
+					$href = null;
```
lines 104-129 indent. then indent this entire section in the same location as the previous code.

alikon commented:
can be a nice addition for 3.9 imo
after a quick test with one of mine  very old package some feedback:
![screenshot from 2018-03-20 19-44-37](https://user-images.githubusercontent.com/181681/37676079-34c1d43e-2c77-11e8-8f36-226844626836.png)

you should add the "linkable" features to modules too

alikon commented on libraries/joomla/form/fields/packagelist.php:
```diff
@@ -0,0 +1,116 @@
+		if (!empty($this->extensiontype))
+		{
+			$query->where('EXISTS ( ' .
+				'SELECT NULL FROM #__extensions AS b ' .
```
why not `SELECT 1` instead of `SELECT NULL`

BrainforgeUK replied:
Use of NULL - habit! Unaware of any performance or other benefit.
This includes a good explanation - even suggests EXISTS ( SELECT 1/0 FROM MYTABLE ) works!
https://dba.stackexchange.com/questions/159413/exists-select-1-vs-exists-select-one-or-the-other

alikon replied:
yeah personal taste ;)

okonomiyaki3000 committed:
Two new fonts for CodeMirror: IBM Plex Mono, Nanum Gothic Coding (#20017)

Commit:
CategoryEdit field published filter (#20018)

infograf768 committed:
Smart Search: Highlighting terms also in fulltext when using readmore (#20019)

* Smart Search: Highlighting terms also in fulltext when using readmore

* parsing summary + body to get text only

SniperSister committed:
Escape full query in NestedTable debug mode (#20024)

SniperSister committed:
Changed viewname filter in RouteHelper (#20031)

Commit:
Fix GMail plugin so it doesn't crash and burn on 4.0 upgrades (#20043)

Commit:
Tweak build script for added flexibility (#19848)

brianteeman committed:
Refresh Manifest Cache failed: Extension is not currently installed (#19560)

* Refresh Manifest Cache failed: Extension is not currently installed

PR for #17604

Change the message to include the name of the extension.

I have no idea how to test this - sorry - only code review - unless someone knows how?

* partial revert

* revert comment

OctavianC committed:
Remove rtrim() since it allows invalid emails (#20080)

Ruud68 committed:
Custom Fields toggle display on read only rights (#20068)

* [com_fields] Normalise the request com_fields data (#19884)

* Normalise the request com_fields data

* CS

* PHP 5.3 compat

* Fields in com_fields array (#9)

Fields should be set in com_fields array and not direcly in $data

* Spelling

* Also normalise request data on front-end user profile save (#10)

* Also normalise request data on front-end user profile save

* correct context and option

* Handle 0 properly in empty check

* Simplify

* allowing value 0 to be saved (#11)

when setting a value of 0 in a text field the function empty will return true > setting the value to null

* correct needsUpdate when strlen (or count) = 1 which incorrectly equa… (#12)

* correct needsUpdate when strlen (or count) = 1 which incorrectly equaled to 'true'

* Update field.php

* Update field.php

* Custom fields view on form via toggle on read-only rights

* fix back-end new article

* first / seperate check on read-only access

* refactor code so show_on parameter is part of helper function

* implement inherit value in fields + language things

* loadmodel only when needed

* changed function comment

* change values order so default value (inherit) is displayed first

* Must use self:: for local static member reference

ggppdk committed:
Fixed page with multiple codemirror editors fields with different syntax highlighting (#20063)

Commit:
Fix for: Can't choose module using editor plugin if you search first (#20005)

* fixit

* cs

* Update modal.php

tonypartridge committed:
Basic check to make sure the bulk import seperator is being used. (#19982)

* Basic check to make sure the bulk import seperator is being used.
Added Import State function as to how the urls should be imported, enabled or disabled.

* force int.

* Update config.xml

* Update links.php

* Update en-GB.com_redirect.ini

* Update config.xml

* Update links.php

* Update en-GB.com_redirect.ini

* Update config.xml

As per standards i.e:
https://github.com/joomla/joomla-cms/blob/staging/administrator/components/com_config/model/form/application.xml

i.e. endtag inline with options and closing tag inline with opening tag.

* Update links.php

tonypartridge committed:
Changed none selected to none, to be used when there are none availab… (#19977)

* Changed none selected to none, to be used when there are none available to select and when none are selected.
Set select to be readonly is they cannot select any options

* Update plugins.php

* Update plugins.php

* Update en-GB.ini

* Update en-GB.ini

* Update plugins.php

* Update plugins.php

* Update plugins.php

* Update plugins.php

* Update plugins.php

Space/tabbing for drone.

* Update plugins.php

emmanuel-tsalampounis committed:
Corrected bug on empty subject of com_mailto (#19956)

* Corrected bug on empty subject

If the subject is empty, the posted value is an empty string (exists) so the default value is never added.

* Updated code to include null value

mino182 committed:
text corrections (#20111)

brianteeman committed:
Typo and copy paste error (#20123)

Someone couldn't spell and then someone else must have copy pasted the error

No idea how to test but this has been wrong since 3.5

zero-24 committed:
correct the use of the use command and move it below the defined command (#20130)

Commit:
Prepare 3.8.7 RC

Commit:
Reset for dev

Commit:
Added trigger event onContentValidateData - See issue 19584

Commit:
Added trigger event onContentValidateData (#19584)

Quy commented on administrator/components/com_plugins/models/forms/filter_plugins.xml:
```diff
@@ -36,6 +36,15 @@
+			type="packagelist"
+			extensiontype="plugin"
+			onchange="this.form.submit();"
+		>
```
Indent.

Quy commented on administrator/components/com_installer/views/manage/tmpl/default.php:
```diff
@@ -100,7 +100,38 @@
+									case 'package':
+										if (!empty($item->search_package_id) && $item->search_package_id == $item->extension_id)
+										{
+												// No link to current search package
```
Reduce indent.

Quy commented on administrator/components/com_plugins/models/plugins.php:
```diff
@@ -263,6 +268,14 @@ protected function getListQuery()
 
+		// Filter by package id.
+		$packageId = $this->getState('filter.package_id');
+		if ($packageId != '')
```
Add blank line above.

Quy commented on libraries/joomla/form/fields/packagelist.php:
```diff
@@ -0,0 +1,116 @@
+		{
+			$query->where('a.enabled = ' . $db->quote($this->packageenabled));
+		}
+		$db->setQuery($query);
```
Add blank line above.

Quy commented on libraries/joomla/form/fields/packagelist.php:
```diff
@@ -0,0 +1,116 @@
+		$packages = $db->loadObjectList();
+
+		$options = array();
+		foreach ($packages as $package)
```
Add blank line above.

Quy commented on libraries/src/Model/Form.php:
```diff
@@ -352,7 +352,14 @@ public function validate($form, $data, $group = null)
 		$dispatcher = \JEventDispatcher::getInstance();
-		$dispatcher->trigger('onUserBeforeDataValidation', array($form, &$data));
+		$calledClass = get_called_class();
+		if (strpos($calledClass, 'User') === 0) {
```
Add blank line above and move `{` to its own next line.

Quy commented on libraries/src/Model/Form.php:
```diff
@@ -352,7 +352,14 @@ public function validate($form, $data, $group = null)
+			// See Joomla Issue #19584
+			$dispatcher->trigger('onUserBeforeDataValidation', array($form, &$data));
+		}
+		// Trigger event name compatible with others in this class
```
Add blank line above.

laoneo commented:
Can you rebase that branch for Joomla 4?

Commit:
Undo previous change.

Commit:
Improved code appearance (blank lines / tabs)

BrainforgeUK commented:
Joomla 4 - I've tried to do a pull request like this.
https://github.com/joomla/joomla-cms/pull/20152

I'm using PHPStorm, and tried rebasing to Joomla 4 as suggested, but unsuccessful.

brianteeman committed:
Add a security policy (#20163)

* Add a security policy

Many projects now add a SECURITY.md document to their repository. Often this is related to using HackerOne but not always.

This PR adds a policy to our github repo. It is based on the existing policy on the d.j.o web site

The file doesn't need to be distributed so it has been added to the exclude list in the github repo.

* tweek

* copy paste

* Update SECURITY.md

* Update SECURITY.md

Commit:
Prepare 3.8.7 release

Commit:
Reset for dev

Commit:
Introduce CODEOWNERS (#20137)

milux committed:
Tidy writeDynaList() (#12184)

* Cleaned writeDynaList() in core.js

* Removed explanation comments

* removed all API changes

* updated compressed core.js

phproberto committed:
[fix] publish/unpublish does not work with tables using null as default checked_out value (#20204)

mansourcodes committed:
Fix overwrite by .table-striped (#20180)

Fix overwrite by administrator/templates/isis/css/template.css line 1787

table.table-striped tbody > tr:nth-child(odd) > td,
table.table-striped tbody > tr:nth-child(odd) > th {
	background-color: #f9f9f9;
}

mansourcodes committed:
Fix overwrite by .table-striped (#20179)

Fix overwrite by administrator/templates/isis/css/template.css line 1787

table.table-striped tbody > tr:nth-child(odd) > td,
table.table-striped tbody > tr:nth-child(odd) > th {
	background-color: #f9f9f9;
}

brianteeman committed:
Tooltips not loading com_users (#20177)

The edit profile form is not loading the bootstrap tooltip code. So any tooltip (not popovers) are displayed as html as seen in the screenshot below when TFA is enabled.

This was spotted by @o2tsen and @sandewt while testing #20051 but as it is a bug effecting more than that PR I have created a new PR. (a pr should only fix one problem)

brianteeman committed:
[a11y] Headings consecutive order Debug Console (#20167)

> Headings communicate the organization of the content on the page. Web browsers, plug-ins, and assistive technologies can use them to provide in-page navigation.

> Skipping heading ranks can be confusing and should be avoided where possible: Make sure that a < h2> is not followed directly by an < h4>, for example.

Source (https://www.w3.org/WAI/tutorials/page-structure/headings/)

The headings were probably chosen for cosmetic reasons and not structural reasons which they should have been

This PR changes the heading in the debug console from h1 to h2

There is a very small visual change as a result but imho the benefits outweigh the small cost

brianteeman committed:
[a11y] Headings consecutive order (#20166)

* [WIP] [a11y] Headings consecutive order

> Headings communicate the organization of the content on the page. Web browsers, plug-ins, and assistive technologies can use them to provide in-page navigation.

> Skipping heading ranks can be confusing and should be avoided where possible: Make sure that a <h2> is not followed directly by an <h4>, for example.

Source (https://www.w3.org/WAI/tutorials/page-structure/headings/)

This PR changes the heading in the plugin and modules from h3 to h2 and in the template styles to h4

### todo
joomla.edit.item_title layout uses h4 but before I change it I need to check everywhere that it is being used

* layout

joomdonation committed:
Fix typos in InstallerControllerUpdate (#20154)

* Fix typos in InstallerControllerUpdate

* Fix same error on other places. Thanks @Quy

* Remove similar unnecessary code

* Revert "Remove similar unnecessary code"

This reverts commit 56410c0f59929d6a5afe220763fed6cfbcda6cdc.

* One more

* Revert "One more"

This reverts commit aa1b101f26be135b7161b4d71aa466f7556e5cab.

Commit:
[com_contact] Don't hide contact filter form (#20126)

* Update default_items.php

* Correct implode order.

* Codestyle

* More codestyle

Commit:
Fix for JUserHelper::addUserToGroup() when user group title is a number. (#20091)

* Update UserHelper.php

* Update UserHelper.php

Commit:
Fix count() in PHP 7.2 (#20044)

alikon committed:
[com_content][Multilanguage] - remove duplicated queries (#19683)

* [com_content][Multilanguage] - remove duplicated queries

* cs

* add $db->qn()

* removed ()

okonomiyaki3000 committed:
Make CodeMirror work in repeatable subforms (#12542)

* One function to initialize any and all CodeMirror instances rather than individual functions to initialize one-by-one. Call on page load and also on subform-row-add

* Minor js changes

* Codemirror fullscreen modifier message (do we still need this?)

okonomiyaki3000 committed:
Call the popover init function when creating new subform rows. (#20222)

* Call the popover init function when creating new subform rows.

* Update teh popover test

brianteeman committed:
[a11y] post-installation message in control panel (#20220)

> Headings communicate the organization of the content on the page. Web browsers, plug-ins, and assistive technologies can use them to provide in-page navigation.

> Skipping heading ranks can be confusing and should be avoided where possible: Make sure that a < h2> is not followed directly by an < h4>, for example.

Source (https://www.w3.org/WAI/tutorials/page-structure/headings/)

The heading was probably chosen for cosmetic reasons and not structural reasons which they should have been

This PR changes the heading for the post-installtion message i the control panel from h4 to h3

There is a very small visual change as a result but imho the benefits outweigh the small cost

degobbis committed:
Solves issue #20195 (#20214)

alikon committed:
[plugin][search][content] give priority on result when title is matched (#20197)

* [plugin][search][content] give priority on result when title is matched

* Missed comma

* Add relevance weighting according to number of words

* Relevance by number of words in title only, removed introtext relevance

* Fix order string concatenation

brianteeman committed:
You've Got Mail (#20162)

* You've Got Mail

Since 2003 the internet has changed. We no longer get a message to say that we have a message. Instead we just give you the message. You probably never use the messages component (especially for private message to a specific user) as they are the equivalent of https://www.youtube.com/watch?v=gFBLiHpkcOk

The Joomla com_messages component is used in two instances

1. Notification of a new article
2. Sending a message to another user

### Current email for Notification of a new article
Subject: A new private message has arrived from [sitename]
Body:
> Please log in to [link] to read your message.

### New email for Notification of a new article
Subject: New message from [user] at [sitename]
Body:
> New Article
A new Article has been submitted by 'user' entitled 'blog post'.
> Please log in to [link] to read your message.

### Current email when sending a message to another user
Subject: A new private message has arrived from [sitename]
Body:
> Please log in to [link] to read your message.

### New email when sending a message to another user
Subject: New message from [user] at [sitename]
Body:
> [subject]
 [message]
[login link]

## Backwards Compatibility
No issues. The message contains the old login message PLUS the content of the message. So if you were using this message in a custom workflow there is no change required to that workflow

* subj

* cs

* add new string and mark existing string for deprecation

okonomiyaki3000 committed:
Support Codemirror's included key mappings (#19833)

* Support Codemirror's included key mappings

* Use a list instead of radio buttons

HLeithner committed:
Don't expose LDAP authentication usage. (#18531)

* Don't expose LDAP authentication usage.

* Use new language strings for LDAP authentication.

* remove bind string

* remove bind string

* use connect string

* alpha order

* alpha order

okonomiyaki3000 committed:
Handle the case that JFolder::files returns 'false' (#11715)

okonomiyaki3000 committed:
Initialize tooltips when a new a row is added in a subform (#12996)

* Initialize tooltips when a new a row is added in a subform

* Fix a test since the init function has changed

* Replace htaccess which was removed inexplicably

Commit:
Missing space (#20260)

Commit:
Tiny JLanguage::loadLanguage() code improvement (#20257)

Commit:
[com_content] Remove redundant check (#20254)

Commit:
Update articles.php (#20245)

Commit:
[com_config] Capitalize label (#20299)

zero-24 committed:
Implement Issue Templates as discussen in #20298

https://github.com/joomla/joomla-cms/issues/20298

rvalitov committed:
[fix] openbase_dir processing (#20280)

okonomiyaki3000 committed:
CodeMirror updated to version 5.37.0 (#20269)

Commit:
Use title from menu item (#20267)

zero-24 committed:
Change the defaults for new installs to disable com_mailto in articles (#20266)

* change the defaults for new installs to disable com_mailto in articles

* change more defaults to 0 thanks @quy

zero-24 committed:
Don't enable sending the PW on new installs (#20247)

* disable plaun pw sending per default on new installs

* make sure we have to set a PW when we dont send the plain pw via mail

* chagne the default in the xml to thanks @quy

* update the sample data thanks @quy

* make sure the mail to user does not include the PW too

* Revert "make sure the mail to user does not include the PW too"

This reverts commit 9095819a9d4e6f8828b0556e5d0284e754f3b9c6.

* address comments by @bakual thanks

Septdir committed:
Optimization and fix of multilingual associations and add layouts to com_content links (#20229)

* Revert #19681

* Revert #19683

* Remove addition query and check after #19314

* Add layout to com_content links

* Add layout to com_content article associations

* Add layout to category associations

* add advanced where clause param

* add advanced where clause for com_content article associations

* drone code formatting fix

* drone code formatting fix

* drone code formatting fix

* Line exceeds 150 characters

* PHPCS rules

* Remove parenthesis

* Change queryKey

* Fix typo

* Improve description

Commit:
Add checksum generation to the build script

Commit:
Hide Enabled/Disabled users column headers and be a11y compliant (#20279)

okonomiyaki3000 committed:
Code mirror autofocus issue (#20270)

* Pass the autofocus param to the plugin.

While we're at it, pretty up this function a little.

* Get autofocus from $params, not $this->params.

Also, make sure we only autofocus one CodeMirror per page.

* We won't be needing these things anymore.

* Keep the unused strings.

csthomas committed:
Add a deprecated flag in database query for a few union/unionAll/unionDistinct (#20219)

Commit:
Add showon attribute (#20156)

Commit:
[com_contact] Display 0 value with custom field (#20124)

* Display 0 value

* Display 0 value

eshiol committed:
fix custom fields with tab format (#20023)

okonomiyaki3000 committed:
Show a sample editor on the CodeMirror plugin config page. (#17042)

* Show a sample editor on the CodeMirror plugin config page.

* Call it a 'Preview'

* Reordered attributes as per the Joomla coding standards for xml

zero-24 committed:
hiddenFieldsets & configFieldsets needs to be defined (#16856)

* hiddenFieldsets & configFieldsets needs to be defined

* use getter

Sophist-UK committed:
Fix subform repeat counter. (#19693)

* Fix subform repeat counter.

As normal rows should be zero indexed.

When length === count === 0, first row should be 0.

Note: PHP creates existing rows with zero index - so if min=1, you
currently get row 0 on form load then row 2, 3 etc. as you add them.

* Update unit tests

* Additional test fixes.

* Fix typo in comment.

* Reminify subform-repeatable.js

johansundell committed:
Fixing so that editors-xtd plugins works with com_ajax (#17939)

* Fixing so that editors-xtd plugins works with com_ajax

* Code formatting fix

Commit:
[Plugins] Add filters to manifest files. (#20410)

infograf768 committed:
URL Menu Item Type link should be required. (#20392)

okonomiyaki3000 committed:
Fix up the protostar template.js (#20224)

Don't use unnecessary closure, just use jQuery ready function.
Use javsacript strict mode
use event delegation where possible
Make tooltips and button groups work properly with repeatable subforms

saharin88 committed:
Bootstrap alert compatible (#18909)

* Bootstrap alert compatible

Alert compatibility with new Bootstrap versions.

* Strict comparison

* Bootstrap alert compatible minified file

Alert compatibility with new Bootstrap versions.

Commit:
Add PHP 7.3 polyfill (#20441)

Commit:
Change the Administrator group's filtering to use the default blacklist

Commit:
Add postinstall message explaining revised default settings

Commit:
Prepare 3.8.8 Release Candidate

Commit:
[com_finder] Fix regression #19969 (#20411)

alikon committed:
[com_finder] - fix update inner join syntax for postgres (#19964)

* [com_finder] - fix update inner join syntax for postgres

* drone fix

cs

* quoteName vs qn

* cs

Commit:
Update tag.php (#19951)

There is a bug in getItem, the function parameter is called $pk, but in code it is using $id, so whenever the parameter  is passed it is not used in the function.

Commit:
Reset for dev

Commit:
Typo in nl-NL installation language file (#20460)

Commit:
Update joomla/filter package

Commit:
Update joomla/application package

Commit:
Store plaintext passwords in a separate options key, do not display passwords on summary page

Commit:
Prepare 3.8.8 release

Commit:
When JFeed was moved, old files were never deleted

Commit:
Reset for dev

Commit:
Correctly escape the random image module output (#20533)

zero-24 committed:
Finally com_mailto allow the usage of a captcha by using JForm (#20265)

* finaly com_mailto allow the usage of a captach by using JForm

* remove unused $session variable

* Line ending to LF

* fix typo thanks @brianteeman

* no need to use set and get thanks @mbabker

* expend the popup & fix the auto population thanks @quy

* captcha handling

* fix the captcha check

* commit header check thanks for reporting @brianteeman @mbabaker and for the improved code @quy

* style changes affected hight thanks @brianteeman

* line ending again

* close <fieldset> thanks @quy

* make the iframe a bit bigger  thnks @quy

* commit proposed changes by @quy

* implement suggested improvments

* use <?php echo $field->renderField(); ?> thanks @laoneo

zero-24 committed:
Remove check for a valid form from two places in plugins (#20277)

* If that is not a valid form our system is broken at all

* Typehint the form argument

Commit:
Fallback to integer (#20338)

Commit:
[plg_user_profile] Misc fixes (#20412)

* Update profile.xml

* Update profile.xml

* Update profile.php

* Update tos.php

Commit:
[plg_fields] Manifest cleanup, filters, Editor plugin fix (#20422)

* Update calendar.xml

* Update checkboxes.xml

* Update checkboxes.xml

* Update editor.xml

* Update editor.xml

* Update editor.php

* Update imagelist.xml

* Update imagelist.xml

* Update integer.xml

* Update integer.xml

* Update list.xml

* Update list.xml

* Update media.xml

* Update radio.xml

* Update radio.xml

* Update checkboxes.xml

* Update list.xml

* Update sql.xml

* Update sql.xml

* Update textarea.xml

* Update textarea.xml

* Update url.xml

* Update url.xml

* Update usergrouplist.xml

* Update usergrouplist.xml

* Restore empty value

Commit:
[plg_system] Manifest file filters (#20456)

photodude committed:
[CS] fix some doc comment types (#20522)

* PHPCS2 manual fixes

- Variables passed by reference should not have the `&` prefixed in the doc comment
  - Joomla.Commenting.FunctionComment.MissingParamTag
  - Joomla.Commenting.FunctionComment.ParamNameNoMatch
- correct return statement

* exclude administrator/components/ folder

"Having a "&" in the docblock (whether it's in front of the type or name) is not valid. References are to be expressed solely by the function/method declaration."

so we need to exclude this sniff while we trasition to the PHPCS 2.x version

Commit:
[styles] Fix checkboxes/buttons not clickable under Menu Assignment in Chrome (#20542)

* Remove css

* Remove css

* Remove css rtl

* cs

Devportobello committed:
[Regression] Menu item type url with rel attribute nofollow (#19949)

* Menu item url attribute rel nofollow with target blank

* logic

* cs

* cs fix - thx Quy

* concat

tomekstach committed:
To many elements to index (#13868)

* To many elements to index

I don't know why in the query `#__tags` is twice. Once as a `a` and second time as a `b`. For me it works perfectly when we remove `b`. I had a problem with this - for 1100 tags on my websites, Smart Search Indexer found more than 1233000 items to index.

* Mistake with query

I corrected a bug with calling the `#__tags` table in the query.

mbabker committed:
Update joomla/filter package (#20579 and #20580)

PhilETaylor committed:
Fix folder browsing and file upload that broke in 3.8.8 due to escaping (#20586)

* Fix de-escaping of slash in folder name

* code style

* Update popup-imagemanager.js

brianteeman committed:
standardise contact strings (#20577)

as spotted by @MartijnMaandag

euismod2336 committed:
Enable to change FormData in com_menu (#20313)

* Set as object to allow for alterations

* Cast to object directly

brianteeman committed:
fix typo in installation string (#20607)

It is IN each catefgory not ON each category

INSTL_DEFAULTLANGUAGE_INSTALL_LOCALISED_CONTENT_DESC="If active, Joomla will automatically create one content category per each installed language. Also, one featured article with dummy content will be created in each category."

mbabker committed:
Resync deleted files list back to 1.7.3 tag (#20564)

tonypartridge committed:
Add additional check that view exists before proceeding (#18757)

* Add additional check that view exists before proceeding

We assume the array key exists with view but not ID. I have come across some installations where view does not exist and it causes php warnings. So just check it does exist before actually proceeding further as we do with ID.

* Update legacyrouter.php

* Update legacyrouter.php

brianteeman committed:
Default installation value in configuration.php-dist (#20655)

* Default installation value in configuration.php-dist

The config file to use in manual installs has incorrect advice and paths for tmp and logs as they are referring to server paths and not paths within a joomla installation

* oops

okonomiyaki3000 committed:
CodeMirror updated to 5.38.0 (#20636)

mbabker committed:
Bump Composer dependencies (#20583)

* Bump Composer dependencies

* Bad user data

* Try changing test to fix PHP 5.3 behavior

* Another attempt at fixing this

* Try upstream changes without tag yet

* Back to empty array

* Now with release tag

* Update app package with UA fix

phproberto committed:
[fix] Error decoding JSON data: Syntax error (#20663)

* [fix] Error decoding JSON data: Syntax error

* [test] ensure that empty params do not cause user loading issues

mbabker committed:
Don't try to json_decode() a null value (#20675)

* Don't try to json_decode() a null value

* Add Registry package fix to make sure broken code can continue to work

Denitz committed:
Joomla\CMS\Categories\Categories fix (#20680)

A constructor can't return a value.

photodude committed:
[CS] fix some doc comment types (round 2) (#20647)

* PHPCS2 manual changes

- Doc comment for parameter does not match actual variable name
- Comment closer must be on a new line (just convert to single line comment)
- be more specific with the rules we want to exclude

* fix spaces

Expected only 2 spaces after the longest variable name

* one more fix for doc comment

- Doc comment for parameter does not match actual variable name

SharkyKZ committed:
[Newsfeeds] Use item link instead of guid. (#20717)

* Feed links

* Restore http check for now

* CS

* Some cleanups

brianteeman committed:
Verb wrong tense (#20708)

* Verb wrong tense

Clearly using the present tense is incorrect for an error message and it should be the past tense

* no error

csthomas committed:
Replace UCMType::getType() by UCMType::getTypeByAlias($this->alias) (#20672)

csthomas committed:
JTableAsset::loadByName(): replace two sql queries by one (#20671)

Commit:
Display 404 page if banner not found (#20664)

SharkyKZ committed:
Update strings (#20591)

ggppdk committed:
Fix stale session data wrongly overriding configuration file when editing global configuration (#20590)

* Fix stale session data wrongly overriding configuration file when editing global configuration

* Use $this variable

alikon committed:
[com_menus] - fix php warning (#18471)

brianteeman committed:
Singular not plural (#20751)

* Singular not plural

Obvious correction. sorry I dont know what I did to display the error.

* Update en-GB.mod_sampledata.ini

SharkyKZ committed:
[com_tags] Image caption in Tags view (#20648)

mbabker committed:
Put all build packages in one directory (#20745)

SharkyKZ committed:
[com_tags] Tagged item link in feed (#20723)

* [com_tags] Tagged item feed links

* Nesting error

SharkyKZ committed:
[mod_tags_similar] Link cleanup (#20730)

* [mod_tags_similar] Link cleanup

* CS

brianteeman committed:
Com_finder meta data (#20772)

* Com_finder meta data

com_finder aka smart search was not following any meta data set in the menu item

#### to test

 - setup and configure smart search
 - create a menu item for smart search and set the meta description and meta keywords
 - open the menu item on the front end - check source and the meta data set in the menu is not present
 - do a search
 - check source and the meta data has been generated from the search

Apply this PR
 - open the menu item on the front end - check source and the meta data set in the menu is present
 - do a search
 - check source and the meta data has been generated from the search

* remove copy paste error

photodude committed:
Add UTF-8 encoding to phpcs in .drone.yml (#20769)

panopt committed:
Login to view the article redirect fix (#20732)

Without the view=login parameter JRoute can not resolve the Itemid. 
So the menu item assigned to 'login' is not active.

Seems there is a related hack here:
/com_content/views/article/tmpl/default.php ~line 143

This can also cause problems with third party extensions like sh404sef.

brianteeman committed:
Mouse over edit - rename (#20743)

At the time this feature was introduced we weren't happy with the name "Mouse over edit" but couldn't really think of anything else.

I believe that today the term "inline editing" is far more understood and is a more appropriate name

okonomiyaki3000 committed:
Better mode autoloading. (#20746)

ggppdk committed:
Respect access of editor plugin inside profile form (#20713)

* Respect access of editor plugin inside profile form

* Use value string 'true', '*' instead of integer for useaccess parameter

PhilETaylor committed:
Additional Escaping Of Paths In com_media (#20616)

* Additional Escaping Of Paths In com_media

* Additional Override Escaping

* carefully revert some escaping (security)

* more specific and special reverse encoding of slash

* C&P Error

SharkyKZ committed:
[com_tags] Images in Tagged Items view (#20601)

alikon committed:
[com_contenthistory] - fix for not delete keep forever items (#20430)

* [com_contenthistory] - fix for not delete kepp forever items

* don't stop delete

* cs

* cs removed spaces at end of line

artur-stepien committed:
Fixing com_fields integration in com_contact (#20413)

* Fixing com_fields integration

Custom fields in com_contact currently doesnt work propertly. Chosing a category on which they should appear doesn't work cause fields plugin has no informations about contact catid as not data is passed to form when contact form is displayed. This fixes it.

* Fixing code style

* Removing whitespace, fixing spelling

Commit:
enable mod_sampledata extension

SharkyKZ committed:
[ModuleHelper] owncache param as integer (#20626)

Commit:
Changed return 0 to continue

To prevent the entire function to quit, it just needs to go to the next file.
Fix https://github.com/joomla/joomla-cms/issues/17954

wilsonge committed:
Merge pull request #20805 from conflate-nl/staging

Changed return 0 to continue

infograf768 committed:
Merge pull request #20797 from Bakual/EnableSampleDataModule

Enable mod_sampledata extension

SharkyKZ committed:
[mod_articles_category] Fatal error (#20834)

Commit:
Prepare 3.8.9 Release Candidate

Commit:
Reset for dev

infograf768 committed:
Very small language file cs (#20844)

Commit:
Whitelist allowed global access

Commit:
Bump joomla/input package

Commit:
Use new method for dynamically resolving Input based on request method

Commit:
Prepare 3.8.9 release

Commit:
Reset for development

mbabker committed:
Fix the autoloader for Windows platforms (#20877)

Commit:
Prepare 3.8.10 release

Commit:
Reset for dev, again

mbabker committed:
Don't delete file that is still in use (Fix #20881)

SharkyKZ committed:
[plg_user_profile] Add RTL check to ToS field

infograf768 committed:
FIX: Install Languages Manager language badges wrongly set to not match since upgrade to 3.8.10 (#20906)

* FIX: Install Languages Manager language badges wrongly set to not match
since upgrade to 3.8.10

* Correcting Installed languages too

csthomas committed:
Replace the URL parameter "limitstart=0" with "start=0" if the SEF mode is on (#19452)

* Replace &limitstart=0 to &start=0 if SEF is ON

* Replace URL param start=x to limitstart=x even if start is 0

tonypartridge committed:
Remove non callable array items from field categories (#20093)

* Update default.php

* Update fields.php

* Update default.php

* Update fields.php

* Update fields.php

* Update fields.php

SharkyKZ committed:
[Site Modules] Manifest file filters, part 1 (#20845)

* [Modules] Manifest file filters

* Chosen placeholders

brianteeman committed:
[cs] leading spaces (#20858)

* [cs]leading spaces

While checking #20844 I saw that these strings all have a leading space which should not be there

* oops

SharkyKZ committed:
[Site Modules] Manifest file filters, part 2 (#20856)

* [Site Modules] Manifest file filters, part 2

* Update default.php

* Update default.php

* Update default.php

SharkyKZ committed:
[Site Modules] Manifest file filters, part 3 (#20857)

SharkyKZ committed:
[Admin Modules] Manifest file filters (#20868)

* [Admin Modules] Manifest file filters

* Default values

Hackwar committed:
Fix OpenSearch implementation

SharkyKZ committed:
[mod_articles_category] Showon

SharkyKZ committed:
Add showon to levels

SharkyKZ committed:
Update string

SharkyKZ committed:
Remove CSS

mbabker committed:
Merge pull request #20950 from SharkyKZ/modArticlesCategoryShowon

[mod_articles_category] Showon

mbabker committed:
Merge pull request #20937 from Hackwar/patch-28

Fix OpenSearch implementation

mbabker committed:
Merge pull request #20894 from SharkyKZ/plg_user_profile_rtl

[plg_user_profile] Add RTL check to ToS field

mbabker committed:
Add security issue template

mbabker committed:
Formatting

Commit:
Let the table check the data before storing.

SharkyKZ committed:
[Templates] Escape sitename (#21008)

Commit:
Checking return value and catching whole stuff

Commit:
Updating Bosnian installation language files

infograf768 committed:
Merge pull request #21080 from Bakual/FixSampledataMenuCreation

Fix sampledata menu creation

jeckodevelopment committed:
Updated 3.9-dev and added 3.10-dev (#21097)

brianteeman committed:
Exceptions in Joomla\CMS\Table\Usergroup refer to categories (#21098)

* Exceptions in Joomla\CMS\Table\Usergroup refer to categories

Pull Request for Issue #21092 .

### Summary of Changes
remove todo that is clearly a copy paste error from somewhere else
change exception message to usergroups from catergory

NOTE the first exception` if ($this->id == 0)`
I am not sure if that should even be present - doesn't look like it to me - please advise

* Update Usergroup.php

SharkyKZ committed:
Deprecate sef_advanced strings (#21116)

SharkyKZ committed:
[com_content] Featured articles tag filtering (#21138)

SharkyKZ committed:
Update ModuleHelper.php (#20273)

mbabker committed:
The JPATH_ constants are not magic strings (#21151)

mbabker committed:
Remove if/else in user debug models that can never hit else conditions (#21188)

SharkyKZ committed:
[com_content] Featured articles access filter (#21168)

wojsmol committed:
Media manager doesn't obey relative pathes for video files (#21156)

* correct video url

fixes https://github.com/joomla/joomla-cms/issues/21145#issuecomment-405484769

* fix

csthomas committed:
Speed up regex in emailcloak plugin (#20956)

* Speed up regex in emailcloak plugin

* Fix comment

ciar4n committed:
Fix subform.repeatable-table multi field styling (#20209)

* Fix subform.repeatable-table multi field styling

* Fix staging less/css

* revert previous

eshiol committed:
Update FormField.php (#20168)

ReLater committed:
Update HTMLHelper.php (#19770)

klas committed:
Fix a problem with older Joomla versions language packs registration and update sites rebuild (#16355)

* Fix a problem with older language packs

Language packs installed under older Joomla versions are registered in the database under their package name, not under their name and since rebuild function fails on them.

e.g. package with 
  <name>Afrikaans (South Africa)</name>
  <packagename>af-ZA</packagename>

has name column in #_extensions table set as 	af-ZA and since existing query fails.

* Update updatesites.php

infograf768 committed:
missing single quote (#21213)

infograf768 committed:
[3.x FIX] Getting the correct associations in sidebyside view for contact and newsfeeds (#21180)

* [3.x FIX] Getting the correct associations in sidebyside view for
contact and newsfeeds

* undefined property

* correction in js

SharkyKZ committed:
[com_menus] Remove useglobal (#21095)

julienV committed:
[fix] fixed PluginHelper import of xtd-editors plugins (#17907)

* [fix] fixed PluginHelper import of xtd-editors plugins

* [imp] small adjustment and comment

okonomiyaki3000 committed:
Fix errors exposed by strict (#12544)

* Don't need a closure, jQuery ready function is enough. Use strict.

* Fix errors exposed by strict. Undeclared variables. wrong regex backreference

* Fix some undeclared/wrongly declared variables

* Avoid some redundant calls of $(this)

* Remove some redundant (also wrong) code.

wojsmol committed:
[3.8] Make the text clearer (#20940)

* make the text clealer

Make the PLG_TWOFACTORAUTH_TOTP_STEP3_TEXT text clealer.

* thans @brianteeman

Commit:
Adding lithuan language files.

Commit:
Updating lithuanian language strings for TinyMCE

brianteeman committed:
Appended input fields (#21257)

PR for #12448

## Steps to reproduce the issue

Effects all appended input fields. Eg. Navigate to a single article menu item (Menus -> MainMenu -> Home) and resize the screen below 767px. Input fields extend outside the viewport.

brianteeman committed:
CSS for flags (#21254)

Fix for #13678 reported by @gwsdesk

salbertson committed:
Add "Reviewed by Hound" badge (#21263)

Commit:
Adjust doc block to ensure file is included in patch packages due to accidental deletion with last release's post-update script

Commit:
Prepare 3.8.11 release candidate

Commit:
Reset to dev

Commit:
Prepare 3.8.11 release

Commit:
Bump to 3.8.12-dev

SniperSister committed:
Added security scan tool to CI setup (#20796)

* Added security scan tool to CI setup

* Updated ordering

* fixed command name

jeckodevelopment committed:
Update readme to link to Launch instead of Demo (#21364)

* No more demo

* No more demo

* Free hosting service

* Free Hosting Service

SharkyKZ committed:
[mod_articles_latest/news] Featured articles not shown (#21336)

SharkyKZ committed:
[com_content] Redundant access check (#21169)

* [com_content] Redundant access check

* Cleanup

* Cleanup

mbabker committed:
Exclude components in discover state from list (Replaces #12469) (#21010)

* Update recaptcha.php

* Update recaptcha.php

Changing load order and adding a comment.
Proposed by @piotr-cz

* Simple CS for travis

* Update helper.php

Exclude components in discover state from list

* Update helper.php

Prevent disabled extensions "enabled = 0"

* Update helper.php

okonomiyaki3000 committed:
Fix minicolor initialization in subforms. etc. (#12994)

jeckodevelopment committed:
Create CODE_OF_CONDUCT.md (#21383)

Added our Code of Conduct as suggested by GitHub.

jeckodevelopment committed:
Removing CODE_OF_CONDUCT (#21385)

As requested in https://github.com/joomla/joomla-cms/pull/21383

brianteeman committed:
CoC Fix the english (#21386)

brianteeman committed:
Readme (#21393)

Removes link to jenkins that is no longer used

infograf768 committed:
[New Feature] Multilingual: Propagating existing associations if desired (#21321)

* [New Feature] Multilingual: Propagating existing associations if desired

* cs

* lang change

* Change and add new messages

* lang changes

* Small corrections without effect on resulting funtionnality

* using new API for token

* modifying js function to use Joomla. instead of window.

mbabker committed:
Revert "[New Feature] Multilingual: Propagating existing associations if desired (#21321)" (#21408)

This reverts commit 8bacca95c2247a91230e56ae2d4a96a98e1ca0dc.

csthomas committed:
Fix unit tests for PostgreSQL 10 (#21370)

csthomas committed:
Escaping float value should be non-locale aware (#20643)

* Escaping float value should be non-locale aware

* Fix sqlite query dateAdd and add unit tests

okonomiyaki3000 committed:
Fix errors exposed by strict and also improve js (#20221)

* Don't need a closure, jQuery ready function is enough. Use strict.

* Radio buttons and tooltips work in subforms.

* Make chose colors work in dynamically added Chosen (in new subform rows, for example)

* Combine the two init functions as one. No reason to call two functions here.

csthomas committed:
Repair the update of database schema changes on postgreSQL (#19707)

* Repair the update of database schema changes on postgreSQL

* Add a better regex to split actions by comma

* Remove useless code

* Add a comment to sql file

BrainforgeUK committed:
Changed indents according to XML standards

https://developer.joomla.org/coding-standards/xml.html

rdeutz committed:
build a not on windows group (#21633)

ciar4n committed:
Stick permissions tab on scroll (#21637)

rdeutz committed:
add a note why we have a not-on-windows group

rdeutz committed:
I will use tabs, I will use tabs, I will use tabs, ….

wojsmol committed:
Update CODEOWNERS after github nick chenge (#21688)

wojsmol committed:
Add .editorconfig to this repository (#21686)

okonomiyaki3000 committed:
Set option by mime type, not by mode name (#21515)

SharkyKZ committed:
[com_content] Tags not shown in some cases (#21275)

* [com_content] Tags not shown in some cases

* Featured view

* CS

Commit:
Fix validation messages in reverse order on installation (#20850)

* Reverse error messages

* Move comment

wilsonge committed:
Cleanup batch copy of items (#13501)

* Cleanup batch copy of items

* Fix codestyle

* Banners doesn't use a title field

* Fix merge conflict

* Fix batch copy not copying fields

wilsonge committed:
Better documentation for addAttachment method. Fixes #14285

BrainforgeUK committed:
Change value column in #__fields_values to longtext (#5)

See issue #19618
[com_fields| Joomla custom field data length

brianteeman commented:
You have added completely unrelated code3 to this issue

You must use a different branch and create a different pull request for each issue you are trying to resolve

In your git client revert this latest change, commit and push

Then create a NEW branch and pull from staging
Make your new changes, commit and push
go to https://github.com/joomla/joomla-cms and you will see info that you can create a new pull request from your new branch

BrainforgeUK committed:
Reverted previous value column in #__fields_value change. (#6) (#7)

Pull request error.

yvesh commented on administrator/components/com_installer/helpers/installer.php:
```diff
@@ -104,10 +104,10 @@ public static function getExtensionGroupes()
+			    ->select('DISTINCT folder')
+			    ->from('#__extensions')
+			    ->where('folder != ' . $db->quote(''))
+			    ->order('folder');
```
Any reason for the indentatoin change? (than you have to change others too, like $db..)

BrainforgeUK committed:
Update installer.php

ka3media commented:
I have tested this item :white_check_mark: successfully on c61135430cd05cb01d659103fd97ec4cb2c28de4<br /><br />Tested on J3.9 alpha

works for me as expected<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/17585">issues.joomla.org/tracker/joomla-cms/17585</a>.</sub>

alikon commented on administrator/components/com_installer/helpers/installer.php:
```diff
@@ -184,4 +184,32 @@ public static function getStateOptions()
+	{
+		$db    = JFactory::getDbo();
+		$query = $db->getQuery(true)
+			->select('extension_id, name')
```
please use `$db->quoteName()`

alikon commented on administrator/components/com_installer/helpers/installer.php:
```diff
@@ -184,4 +184,32 @@ public static function getStateOptions()
+		$query = $db->getQuery(true)
+			->select('extension_id, name')
+			->from('#__extensions')
+			->where('type = ' . $db->quote('package'))
```
here

alikon commented on administrator/components/com_installer/helpers/installer.php:
```diff
@@ -184,4 +184,32 @@ public static function getStateOptions()
+			->select('extension_id, name')
+			->from('#__extensions')
+			->where('type = ' . $db->quote('package'))
+			->order('name');
```
here

alikon commented on administrator/components/com_installer/models/manage.php:
```diff
@@ -335,6 +337,12 @@ protected function getListQuery()
 
+		if ($packageId != '')
+		{
+			$query->select((int) $packageId . ' AS search_package_id')
```
here

BrainforgeUK replied:
$packageId is a numeric value, not a column name.

alikon commented on administrator/components/com_plugins/models/plugins.php:
```diff
@@ -263,6 +268,15 @@ protected function getListQuery()
+		if ($packageId != '')
+		{
+			$query->select((int) $packageId . ' AS search_package_id')
+			      ->where('package_id = ' . (int) $packageId);
```
here

BrainforgeUK replied:
$packageId is a numeric value, not a column name.

alikon replied:
`package_id` is a field/column of the `#__extensions` table

alikon commented on libraries/joomla/form/fields/packagelist.php:
```diff
@@ -0,0 +1,118 @@
+	public function getOptions()
+	{
+		$db    = JFactory::getDbo();
+		$query = $db->getQuery(true)
```
here

BrainforgeUK replied:
Changes made.

alikon reviewed (changes requested):
use $db->quoteName()

BrainforgeUK committed:
Update installer.php

BrainforgeUK committed:
Now uses $db->quoteName()

BrainforgeUK committed:
Now uses $db->quoteName()

BrainforgeUK committed:
Added $db->quoteName()

BrainforgeUK committed:
Improved extension update invalid package error handling.

See issue # 22762

BrainforgeUK committed:
Merge branch 'staging' of https://github.com/BrainforgeUK/joomla-cms into staging

Quy commented:
Please rebase for J4. <hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/17585">issues.joomla.org/tracker/joomla-cms/17585</a>.</sub>
